src/app/shared/spaces.service.ts

```diff
@@ -1,72 +1,80 @@
 import { Injectable } from '@angular/core';
-
-import { LocalStorageService } from 'angular-2-local-storage';
 import { Broadcaster } from 'ngx-base';
 import { Contexts, Space, Spaces, SpaceService } from 'ngx-fabric8-wit';
-import { Observable, Subject } from 'rxjs';
-import { ConnectableObservable } from 'rxjs/observable/ConnectableObservable';
-import { ReplaySubject } from 'rxjs/ReplaySubject';
-
+import { ConnectableObservable, Observable, ReplaySubject, Subscription } from 'rxjs';
 import { ExtProfile, ProfileService } from '../profile/profile.service';
 
 @Injectable()
 export class SpacesService implements Spaces {
 
   static readonly RECENT_SPACE_LENGTH = 8;
-  private static readonly RECENT_SPACE_KEY = 'recentSpaceIds';
 
-  addRecent: Subject<Space> = new Subject<Space>();
   private _current: Observable<Space>;
   private _recent: ConnectableObservable<Space[]>;
 
   constructor(
     private contexts: Contexts,
-    private localStorage: LocalStorageService,
     private spaceService: SpaceService,
     private broadcaster: Broadcaster,
     private profileService: ProfileService
   ) {
+    this.initCurrent();
+    this.initRecent();
+  }
+
+  private initCurrent(): void {
     this._current = Observable.merge(
-      contexts.current
-        .map(val => val.space),
-      this.broadcaster.on<Space>('spaceUpdated'));
-    // Create the recent context list
+      this.contexts.current.map(val => val.space),
+      this.broadcaster.on<Space>('spaceUpdated')
+    );
+  }
+
+  private initRecent(): void {
+    // load existing recent spaces from profile.store.recentSpaces
+    this._recent = this.loadRecent().publishReplay(1);
+    this._recent.connect();
```

Comment:
I'm not seeing the need for this to be a `ConnectableObservable`. It's connected immediately after instantiation anyway, during the service's constructor. Does this have any practical effect?